projects/dotcms-field-elements/src/components/dot-textfield/dot-textfield.tsx

```diff
@@ -1,66 +1,134 @@
-import { Component, Prop, State, Event, EventEmitter } from '@stencil/core';
-import { generateId } from '../../utils';
+import { Component, Prop, State, Event, EventEmitter, Method } from '@stencil/core';
 import Fragment from 'stencil-fragment';
 
 @Component({
     tag: 'dot-textfield',
     styleUrl: 'dot-textfield.scss'
 })
 export class DotTextfieldComponent {
-    @Prop() value: string;
+    @Prop({ mutable: true })
+    value: string;
+    @Prop() name: string;
     @Prop() regexcheck: string;
-    @Prop() readOnly: string;
+    @Prop() regexcheckmessage: string;
     @Prop() label: string;
     @Prop() hint: string;
     @Prop() placeholder: string;
     @Prop() required: boolean;
-    @Event() onChange: EventEmitter;
+    @Prop() requiredmessage: string;
+    @Prop() disabled = false;
+    @Event() valueChanges: EventEmitter;
+    @Event() statusChanges: EventEmitter;
 
-    @State() _value: string;
-    @State() _error = false;
-    _label: string;
+    @State() _valid = true;
+    @State() _dotTouched = false;
+    _dotPristine = true;
```

Comment:
Why this is not state also?

Reply:
because `_dotPristine ` will be updated only when `value` change, so `render()` is always executed. `_dotTouched ` can be changed without affecting any `Prorp`